src/main/java/ch/njol/skript/hooks/permission/expressions/ExprPlayerPerms.java

```diff
@@ -0,0 +1,100 @@
+package ch.njol.skript.hooks.permission.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.hooks.VaultHook;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import java.util.Arrays;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.permissions.PermissionAttachmentInfo;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Player Permissions")
+@Description("All permissions of a player")
+@Examples("add \"perms.test\" to permissions of event-player")
+@Since("INSERT VERSION")
+public class ExprPlayerPerms extends SimpleExpression<String> {
+	static {
+		Skript.registerExpression(ExprPlayerPerms.class, String.class, ExpressionType.PROPERTY,
+			"[all] permissions of %player%",
+			"%player%'s [all] permissions");
+	}
```

Comment:
Should conflict with https://github.com/SkriptLang/Skript/blob/ba32d7ad48edb8e33d5f5d2e44d03bba94ffb2de/src/main/java/ch/njol/skript/expressions/ExprPermissions.java